src/hubdata/connect_target_data.py

```diff
@@ -0,0 +1,98 @@
+    :raise: RuntimeError if `hub_path` is invalid
+    :raise: RuntimeError if hub has no `hub-config/target-data.json` file
+    :raise: RuntimeError if hub has no time-series target data, i.e., no `target-data/time-series.csv`,
+        `target-data/time-series.parquet`, or `target-data/time-series/` files/dir
+    """
+    hub_conn = connect_hub(hub_path)  # raises RuntimeError if hub_path is invalid
+    found_file_info = _validate_target_data(hub_conn, True)  # raises RuntimeError if hub has no target data
+    return _create_dataset(hub_conn, found_file_info, True)
+
+
+def connect_target_oracle_output(hub_path: str | Path) -> ds.dataset:
+    """
+    Top-level function for accessing the oracle-output data for the passed `hub_path`.
+
+    :param hub_path: str (for local file system hubs or cloud based ones) or Path (local file systems only) pointing to
+        a hub's root directory. it is passed to https://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html#pyarrow.fs.FileSystem.from_uri
+        From that page: Recognized URI schemes are “file”, “mock”, “s3fs”, “gs”, “gcs”, “hdfs” and “viewfs”. In
+        addition, the argument can be a local path, either a pathlib.Path object or a str. NB: Passing a local path as a
+        str requires an ABSOLUTE path, but passing the hub as a Path can be a relative path.
+    :return: a `ds.dataset` for the passed `hub_path`. note that we return a dataset for the single file cases so that
+        the user can control when data is materialized into memory
+    :raise: RuntimeError if `hub_path` is invalid
+    :raise: RuntimeError if hub has no `hub-config/target-data.json` file
+    :raise: RuntimeError if hub has no oracle-output target data, i.e., no `target-data/oracle-output.csv`,
+        `target-data/oracle-output.parquet`, or `target-data/oracle-output/` files/dir
+    """
+    hub_conn = connect_hub(hub_path)  # raises RuntimeError if hub_path is invalid
+    found_file_info = _validate_target_data(hub_conn, False)  # raises RuntimeError if hub has no target data
+    return _create_dataset(hub_conn, found_file_info, False)
+
+
+def _validate_target_data(hub_conn: HubConnection, is_time_series: bool) -> fs.FileInfo:
+    """
+    Helper that validates target data (either time-series or oracle-output) by looking for the three supported cases:
+    - a csv file (time-series.csv or oracle-output.csv)
+    - a parquet file (time-series.parquet or oracle-output.parquet)
+    - a partitioned directory (time-series/ or oracle-output/ )
+
+    :param hub_conn: the hub's HubConnection
+    :param is_time_series: True if output is for time-series target data, and False if for oracle-output target data
+    :return: the fs.FileInfo of the single found target data file or dir
+    :raise: RuntimeError if hub has no time-series target data file or dir
+    """
+    target_data_name = 'time-series' if is_time_series else 'oracle-output'
```

Comment:
To follow my previous suggestion
```suggestion
def _validate_target_data(hub_conn: HubConnection, target_type = "time-series") -> fs.FileInfo:
    """
    Helper that validates target data (either time-series or oracle-output) by looking for the three supported cases:
    - a csv file (time-series.csv or oracle-output.csv)
    - a parquet file (time-series.parquet or oracle-output.parquet)
    - a partitioned directory (time-series/ or oracle-output/ )

    :param hub_conn: the hub's HubConnection
    :param target_type: str indication type of target data to retrieve matching files. One of "time-series" or "oracle-output". Defaults to "time-series"
    :return: the fs.FileInfo of the single found target data file or dir
    :raise: RuntimeError if hub has no time-series target data file or dir
    """
    if target_type not in ["time-series", "oracle-output"]:
        raise ValueError('{target_type} should be "time-series" or "oracle-output"')
```